bin/import_xml2obd.py

```diff
@@ -236,16 +302,30 @@ def upload_resources(remote_site, api_key, jsonfile, resource_directory):
             continue
         res = rec['resources'][0]
         source = resource_directory + '/' + res['url'].split('/')[-1]
-        with open(source) as f:
-            try:
-                rc = site.action.resource_patch(
-                    id=res['id'],
-                    upload=(os.path.basename(source), f))
-            except ckanapi.errors.CKANAPIError:
-                traceback.print_exc()
-                print(rec)
-                print("failed upload " + os.path.basename(source))
-                continue
+        fname='/'.join(['resources', res['id'], res['url'].split('/')[-1]])
+
+        obj = dest.get_obj(fname.lower())
+        skip = False
+        if obj:
+            objmd5 = base64md5str(obj.properties.content_settings.content_md5)
+            srcmd5 = md5str(source)
+            if objmd5 == srcmd5:
+               print('\tsame remote file exists ',  os.path.basename(source))
+               skip = True
+            else:
+                print (objmd5, srcmd5)
+        if not skip:
+            with open(source) as f:
+                try:
+                    rc = site.action.resource_patch(
+                        id=res['id'],
```

Comment:
add `hash='md5-%s' % obj.properties.content_settings.content_md5,` here to store the hash in the resource too